src/main/scala/tutorial/lesson1-circuit-traversal/AnalyzeCircuit.scala

```diff
@@ -0,0 +1,140 @@
+package tutorial
+package lesson1
+
+// Compiler Infrastructure
+import firrtl.{Transform, LowForm, CircuitState}
+// Firrtl IR classes
+import firrtl.ir.{Circuit, DefModule, Statement, Expression, Mux}
+// Map functions
+import firrtl.Mappers._
+// Scala's mutable collections
+import scala.collection.mutable
+
+/** Ledger tracks [[Circuit]] statistics
+  *
+  * In this lesson, we want to count the number of muxes in each
+  *  module in our design.
+  *
+  * This [[Ledger]] class will be passed along as we walk our
+  *  circuit, and help us count each [[Mux]] we find.
+  *
+  * See [[lesson1.AnalyzeCircuit]]
+  */
+class Ledger {
+  private var moduleName: Option[String] = None
+  private val modules = mutable.Set[String]()
+  private val moduleMuxMap = mutable.Map[String, Int]()
+  def foundMux: Unit = moduleName match {
+    case None => error("Module name not defined in Ledger!")
+    case Some(name) => moduleMuxMap(name) = moduleMuxMap.getOrElse(name, 0) + 1
+  }
+  def getModuleName: String = moduleName match {
+    case None => error("Module name not defined in Ledger!")
```

Comment:
This error method is deprecated, should use sys.error or add an import if something else is wanted